R/DMMParam.R

```diff
@@ -0,0 +1,203 @@
+#' DMM clustering
+#' 
+#' Run the \code{DMM} function on a distance matrix 
+#' within \code{\link{clusterRows}}.
+#' 
+#' @param k An integer indicating the number of clusters, or a list of integers 
+#'   defining the possible number of clusters to choose from.
+#' @param type A string specifying the fit function to use to find the optimal 
+#'   number of clusters. Use in combination with k being a list of integers. 
+#'   Must be equal to \code{laplace}, \code{AIC} or \code{BIC}.
+#' @param transposed Logical scalar, is x transposed with samples in rows?
+#' @param seed An integer specifying the seed to use when doing the DMM 
+#'   algorithm. Will be random if not set.
+#' 
+#' @inheritParams clusterRows
+#' 
+#' @author Basil Courbayre
+#'
+#' @references
+#' Ian Holmes, Keith Harris, Christopher Quince (2012)
+#' Dirichlet multinomial mixtures: generative models for microbial metagenomics.
+#' \url{https://pubmed.ncbi.nlm.nih.gov/22319561/}
+#' 
+#' @details
+#' The DMM algorithm (see Holmes et al. 2012) is commonly used in microbial ecology along with metagenomic & 16S rRNA count data.
+#' 
+#' If we have microbiota data (for example the Twins data set in the DMM package), the suitable algorithm can be DMM.
+#' To modify an existing DMMParam object \code{x},
+#' users can simply call \code{x[[i]]} or \code{x[[i]] <- value} where \code{i} is any argument used in the constructor.
+#'
+#' If \code{k=NULL}, the cluster search range will be 1:3.
+#' The default for the fit function is \code{laplace}.
+#' 
+#' @return 
+#' The \code{DMMParam} constructor will return a \linkS4class{DMMParam} object with the specified parameters.
+#'
+#' The \code{clusterRows} method will return a factor of length equal to \code{nrow(x)} containing the cluster assignments.
+#' If \code{full=TRUE}, a list is returned with \code{clusters} (the factor, as above) and \code{objects} 
+#' (a list containing \code{dmm}, the dmm object; \code{k}, the number of clusters; 
```

Comment:
What is the "dmm object"? If it's too hard to describe, just say `the output of \code{\link[DirichletMultinomial]{dmn}}`.